### PR TITLE
feat: setup react page with submenu for datasources listview 

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

- Sets up infra for rendering a react view under `databaseview/list/` when `ENABLE_REACT_CRUD_VIEWS` config flag is enabled and `SIP_34_DATABASE_UI` feature flag is enabled. 
- adds a submenu to the database list page. 
- moves props shared between datasets and databases subMenu into a common file.

- renames the "welcome" app to "crud_views" and moves the welcome view under `views/CRUD/` 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2020-08-19 at 11 21 03 AM

### TEST PLAN

- FAB view renders when `ENABLE_REACT_CRUD_VIEWS` and `SIP_34_DATASOURCE_UI` are both off.
- React view (only showing SubMenu) renders when `ENABLE_REACT_CRUD_VIEWS` and `SIP_34_DATASOURCE_UI` are both on.

- unit tests for DatasourceList
- CI
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by kalimuthu123 — COMMENTED** [review]
good processs

File: superset-frontend/src/views/CRUD/dataset/DatasetList.tsx [hunk 1]
```diff
@@ -448,7 +448,11 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
         label: t('Datasets'),
         url: '/tablemodelview/list/',
       },
-      { name: 'Databases', label: t('Databases'), url: '/databaseview/list/' },
+      {
+        name: 'Dasources',
```
Comment (riahk) [inline superset-frontend/src/views/CRUD/dataset/DatasetList.tsx:452]:
```suggestion
        name: 'Datasources',
```

**Review by riahk — COMMENTED** [review]
It's probably not in scope (especially with how routing is currently setup), but we should be aiming to get all the lists within data to share a `SubMenu` to avoid repeating it in each list view.

File: superset-frontend/src/views/CRUD/dataset/DatasetList.tsx (hunk 1 above)
Comment (nytai) [reply superset-frontend/src/views/CRUD/dataset/DatasetList.tsx]: thanks for catching that!

**Comment by nytai** [comment]
@riahk I agree, this copying of the subMenu around is less than ideal. You very much identified the challenge, the way routing is set up. Ideally routing would look something like `/data/datasources` and `/data/datasets` and the SubMenu would live under the `/data` route portion. What's preventing that change is that we're currently feature flagging these views and have to support legacy routes. Once we fully deprecate the legacy views (and routes, potentially) we can set up permanent redirects.

**Comment by mistercrunch** [comment]
There are really "Databases". The term "datasource" ambiguated with "dataset". Let's use "Databases". Maybe we need a short glossary in the docs.

File: superset-frontend/src/views/CRUD/dataset/DatasetList.tsx [hunk 2]
```diff
@@ -448,7 +448,11 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
         label: t('Datasets'),
         url: '/tablemodelview/list/',
       },
-      { name: 'Databases', label: t('Databases'), url: '/databaseview/list/' },
+      {
+        name: 'Datasources',
+        label: t('Datasources'),
+        url: '/databaseview/list/',
```
Comment (mistercrunch) [inline superset-frontend/src/views/CRUD/dataset/DatasetList.tsx:454]: I vote for moving to `/database/list/`. The "view" suffix is a FAB artifact from what the ModelView class was class.

**Comment by nytai** [comment]
@riahk took a shot at sharing a menu component between Datasets and Databases, but quickly realized that only half the menu is shared. There's logic that conditionally adds the "Bulk Select" and "+Dataset" actions, those are specific to the Datasets view. Instead I just moved the props that are shared into a common file. 

I also renamed the "Welcome" app to something more semantic and moved the Welcome component under `src/views/CRUD`

File: superset-frontend/src/components/Modal.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ import React from 'react';
 import styled from '@superset-ui/style';
 import { Modal as BaseModal } from 'react-bootstrap';
 import { t } from '@superset-ui/translation';
-import Button from 'src/views/CRUD/dataset/Button';
+import Button from 'src/views/CRUD/data/dataset/Button';
```
Comment (rusackas) [inline superset-frontend/src/components/Modal.tsx:23]: FWIW, I'm planning to open a PR soon sweeping through and cleaning out any one-off buttons, including this fella. One button to rule them all! ;) I know this is the newest-looking button, so I'll be supporting various bits and pieces from it.

File: superset-frontend/src/views/App.tsx [hunk 4]
```diff
@@ -30,14 +30,15 @@ import Menu from 'src/components/Menu/Menu';
...
-import setupApp from '../setup/setupApp';
-import setupPlugins from '../setup/setupPlugins';
-import Welcome from './Welcome';
-import ToastPresenter from '../messageToasts/containers/ToastPresenter';
+import messageToastReducer from 'src/messageToasts/reducers';
+import { initEnhancer } from 'src/reduxUtils';
+import setupApp from 'src/setup/setupApp';
+import setupPlugins from 'src/setup/setupPlugins';
+import Welcome from 'src/views/CRUD/welcome/Welcome';
+import ToastPresenter from 'src/messageToasts/containers/ToastPresenter';
```
Comment (rusackas) [inline superset-frontend/src/views/App.tsx:41]: thanks for catching these... they're _everywhere_

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx [hunk 5]
```diff
@@ -21,8 +21,8 @@ import { Panel, Row, Col, Tabs, Tab, FormControl } from 'react-bootstrap';
 import { t } from '@superset-ui/translation';
 import { useQueryParam, StringParam, QueryParamConfig } from 'use-query-params';
 import { User } from 'src/types/bootstrapTypes';
-import RecentActivity from '../profile/components/RecentActivity';
-import Favorites from '../profile/components/Favorites';
+import RecentActivity from '../../../profile/components/RecentActivity';
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:24]:
```suggestion
import RecentActivity from 'src/profile/components/RecentActivity';
```

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx [hunk 6]
```diff
@@ -21,8 +21,8 @@ import { Panel, Row, Col, Tabs, Tab, FormControl } from 'react-bootstrap';
...
+import Favorites from '../../../profile/components/Favorites';
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:25]:
```suggestion
import Favorites from 'src/profile/components/Favorites';
```

File: superset-frontend/webpack.config.js [hunk 7]
```diff
@@ -180,7 +180,7 @@ const config = {
     explore: addPreamble('/src/explore/index.jsx'),
     dashboard: addPreamble('/src/dashboard/index.jsx'),
     sqllab: addPreamble('/src/SqlLab/index.tsx'),
-    welcome: addPreamble('/src/welcome/index.tsx'),
+    crud_views: addPreamble('/src/views/index.tsx'),
```
Comment (rusackas) [inline superset-frontend/webpack.config.js:183]: any reason not to follow the camelCase pattern here?

File: superset-frontend/src/components/Modal.tsx (hunk 3 above)
Comment (nytai) [reply superset-frontend/src/components/Modal.tsx:23]: Yes, the new buttons are this one and the really wide CTA one. Maybe they can just be variants of the `bsSize` attribute

File: superset-frontend/webpack.config.js (hunk 7 above)
Comment (nytai) [reply superset-frontend/webpack.config.js]: Sine these are consumed on the python side, I thought snake case made sense. Realizing now that there's another bundle that's camelCased

File: superset-frontend/src/components/Modal.tsx (hunk 3 above)
Comment (rusackas) [reply superset-frontend/src/components/Modal.tsx:23]: right... maybe a "cta" prop... to apply these styles would do just fine. It just needs a good name, really :D Anyway, we can discuss this outside of this PR.

**Review by rusackas — APPROVED** [review]
Looks like there are a few nits questions remaining that sound like they're being addressed, but I don't see any deal-breakers, so LGTM!

File: superset-frontend/webpack.config.js (hunk 7 above)
Comment (nytai) [reply superset-frontend/webpack.config.js]: switched to camelCase for consistency

**Pull request merged**